src/libpmem/pmem_posix.c

```diff
@@ -55,6 +55,9 @@ is_pmem_detect(const void *addr, size_t len)
 {
 	LOG(3, "addr %p len %zu", addr, len);
 
+	if (len == 0)
+		return 0;
```

Comment:
`len = 1;`
As is, this will return false for an address that is potentially pmem. And I think people might want to use `pmem_is_pmem` with 0 len. This often comes up in loops, where you traverse something and subtract a size - making a special case for 0 is always annoying.
Same below.

Reply:
Applications calling pmem_is_pmem with length==0 and expecting result for length==1 are buggy. And I can't imagine any application that would call pmem_is_pmem in a loop for contiguous piece of memory. We even document it to be slow.

This change is not about providing meaningful result (because there's none), but about consistent behavior between supported operating systems.